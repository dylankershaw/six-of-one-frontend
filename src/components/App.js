import React, { Component } from "react";

import logo from "../assets/logo_and_hamburger.svg";
import coverPhoto from "../assets/cover_photo.jpg";
import Description from "./Description";
import ReadTitle from "./ReadTitle";
import EditTitle from "./EditTitle";
import Body from "./Body";

class App extends Component {
  constructor() {
    super();

    this.state = { mode: "read" };
  }

  render() {
    return (
      <div>
        <img className="logo" alt="six of one logo" src={logo} />
        <img
          className="cover-photo"
          alt="mountain landscape"
          src={coverPhoto}
        />
        {this.state.mode === "read" ? (
          <ReadTitle changeMode={this.changeMode} />
        ) : (
          <EditTitle changeMode={this.changeMode} />
        )}
        <Description />
        <Body />
      </div>
    );
  }

  changeMode = mode => {
    this.setState({ mode });
  };
}

export default App;
