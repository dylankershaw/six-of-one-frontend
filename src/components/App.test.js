import React from "react";
import { shallow } from "enzyme";

import App from "./App";

describe("App", () => {
  const app = shallow(<App />);

  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });

  it("initializes the `state` in read mode", () => {
    expect(app.state().mode).toEqual("read");
  });
});
