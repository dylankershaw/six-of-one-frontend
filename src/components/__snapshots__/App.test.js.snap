// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="six of one logo"
          className="logo"
          src="logo_and_hamburger.svg"
/>,
        <img
          alt="mountain landscape"
          className="cover-photo"
          src="cover_photo.jpg"
/>,
        <withRouter(ReadTitle)
          changeMode={[Function]}
/>,
        <Description />,
        <Body />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "six of one logo",
          "className": "logo",
          "src": "logo_and_hamburger.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "mountain landscape",
          "className": "cover-photo",
          "src": "cover_photo.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeMode": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="six of one logo"
            className="logo"
            src="logo_and_hamburger.svg"
/>,
          <img
            alt="mountain landscape"
            className="cover-photo"
            src="cover_photo.jpg"
/>,
          <withRouter(ReadTitle)
            changeMode={[Function]}
/>,
          <Description />,
          <Body />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "six of one logo",
            "className": "logo",
            "src": "logo_and_hamburger.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "mountain landscape",
            "className": "cover-photo",
            "src": "cover_photo.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeMode": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
